File: config/initializers/typesense.rb
Typesense.configuration = {
  nodes: [{
    host: ENV["TYPESENSE_HOST"] || "localhost",
    port: ENV["TYPESENSE_PORT"] || 8108,
    protocol: ENV["TYPESENSE_PROTOCOL"] || "http",
  }],
  api_key: ENV["TYPESENSE_API_KEY"] || "xyz",
  connection_timeout_seconds: 2000,
  retry_interval_seconds: 3,
  num_retries: 5,
  logger: Logger.new($stdout),
  log_level: Logger::DEBUG,
}

File: app/frontend/lib/typesense.ts
import type { ConfigurationOptions } from "typesense/lib/Typesense/Configuration";
import { Client } from "typesense";

const config = {
  TYPESENSE_HOST: "localhost",
  TYPESENSE_PORT: 8108,
  TYPESENSE_PROTOCOL: "http",
  TYPESENSE_API_KEY: "xyz",
};

export const typesenseConfig: ConfigurationOptions = {
  nodes: [
    {
      host: config.TYPESENSE_HOST,
      port: Number(config.TYPESENSE_PORT),
      protocol: config.TYPESENSE_PROTOCOL,
    },
  ],
  apiKey: config.TYPESENSE_API_KEY,
  retryIntervalSeconds: 2,
};

export const client = new Client(typesenseConfig);
